app/models/user_track.rb

```diff
@@ -13,40 +13,38 @@ def self.for!(user_param, track_param)
     )
   end
 
-  def self.for(user_param, track_param)
+  def self.for(user_param, track_param, external_if_missing: false)
     for!(user_param, track_param)
   rescue ActiveRecord::RecordNotFound
-    nil
-  end
+    return nil unless external_if_missing
 
-  def solutions
-    user.solutions.joins(:exercise).where("exercises.track_id": track)
+    begin
+      External.new(Track.for!(track_param))
+    rescue ActiveRecord::RecordNotFound
+      nil
+    end
   end
 
-  # A track's summary is a effeciently created summary of all
-  # of a user_track's data. It's cached across requests, allowing
-  # us to quickly retrieve data without requiring lots of complex
-  # SQL queries. There is a little bit of a dance here, which is
-  # documented in the UserTrack::GenerateSummary class.
-  attr_writer :summary
-  def summary
-    @summary ||= UserTrack::GenerateSummary.(track, self)
+  def external?
+    false
   end
 
-  def summary_generated?
-    !!@summary
+  def solutions
+    user.solutions.joins(:exercise).where("exercises.track_id": track)
   end
 
-  delegate :exercise_available?, :concept_available?, to: :summary
+  delegate :exercise_available?, :exercise_completed?,
+    :num_completed_exercises,
+    :num_concepts, :num_concepts_mastered,
+    :num_exercises_for_concept, :num_completed_exercises_for_concept,
+    :concept_available?, :concept_mastered?,
+    to: :summary
```

Comment:
This whole commit is basically this.

Reply:
I love how there is stuff like this built-in :D

Reply:
Yeah. It's good huh. This is Rails, not Ruby btw.